doc/sphinx/conf.py

```diff
@@ -317,42 +311,53 @@
 if all(os.path.exists(file) for file in cppguide_files):
     html_extra_path.extend(str(file) for file in cppguide_files)
 
-
-def read_doxyfile(path):
-    # Instead of configuring Mir again to get the doxyfile, we assume the user
-    # already has configured it and grab the file from there.
-    try:
-        # Read the doxyfile from the build directory
-        with open(path) as doxyfile_file:
-            return doxyfile_file.read()
-    except IOError as e:
-        raise Exception(f"""IOError: {e.errno}, {e.strerror}
-                        \rHint: Have you configured the cmake project and changed `cmake_build_dir` to point at it?""")
-
-
-doxyfile_contents = read_doxyfile(cmake_build_dir / 'doc/sphinx/Doxyfile')
-
-# Setup the exhale extension
-exhale_args = {
-    # These arguments are required
-    "containmentFolder": "./api",
-    "rootFileName": "library_root.rst",
-    "doxygenStripFromPath": "..",
-    # Heavily encouraged optional argument (see docs)
-    "rootFileTitle": "Mir API",
-    # Suggested optional arguments
-    "createTreeView": False,
-    "exhaleExecutesDoxygen": True,
-    "exhaleUseDoxyfile": False,
-    "contentsDirectives": False,
-    "exhaleDoxygenStdin": doxyfile_contents
-}
-
-# Setup the breathe extension
-breathe_projects = {"Mir": "./xml/"}
-breathe_default_project = "Mir"
-breathe_default_members = ('members', 'undoc-members')
-breathe_order_parameters_first = True
+build_api_docs = os.getenv('MIR_BUILD_API_DOCS') == '1'
+
+# Only add exhale and breathe if the user ran doxygen (which produces `<build_dir>/doc/sphinx/doxygen_output/xml`)
+if build_api_docs:
+    cmake_build_dir = os.getenv('MIR_CMAKE_BUILD_DIR')
+    if not cmake_build_dir:
+        raise Exception("MIR_CMAKE_BUILD_DIR is not set.\n"
+                        "Hint: Set MIR_CMAKE_BUILD_DIR to the path of the configured Mir cmake project relative to "
+                        "`doc/sphinx/conf.py`, or an absolute path")
+
+    sphinx_dir = Path(cmake_build_dir) / 'doc/sphinx'
+    if not os.path.exists(sphinx_dir):
+        raise Exception(f"Cannot find '{sphinx_dir}'.\n"
+                        "Hint: Set MIR_CMAKE_BUILD_DIR to the path of the configured Mir cmake project relative to "
+                        "`doc/sphinx/conf.py`, or an absolute path")
+
+    doxygen_xml_output_dir = sphinx_dir / "doxygen_output/xml"
+    doxygen_xml_exists = os.path.exists(doxygen_xml_output_dir)
+
+    if build_api_docs and not doxygen_xml_exists:
+        raise Exception(f"Doxygen outputs at {doxygen_xml_output_dir} don't exist\n"
+                        "Hint: Make sure MIR_CMAKE_BUILD_DIR points at the the configured Mir cmake project correctly\n"
+                        "If you've already ran doxygen, a directory named `doxyen_output` should exist inside the build"
+                        " directory")
```

Comment:
I _think_ this can be flattened to a single error, along the lines of:
```suggestion
  try:
    doxygen_xml_output_dir = Path(os.environ["MIR_CMAKE_BUILD_DIR"]) / "doc/sphinx/doxygen_output/xml"
    assert doxygen_xml_output_dir.is_dir(), "Doxygen XML output not found"

  except KeyError, AssertionError as ex:
    raise Exception("MIR_CMAKE_API_DOCS is set, but couldn't find Doxygen output. "
                    "Make sure `MIR_CMAKE_BUILD_DIR` points at the CMake build directory "
                    "and that `doxygen_output` is populated inside\n"
                    f"{ex}") from ex
```